File: Leetcode/1863. Sum of All Subset XOR Totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def helper(nums, pos, XOR):
            return XOR if pos == len(nums) else helper(nums, pos + 1, XOR) + helper(nums, pos + 1, XOR ^ nums[pos])
        return helper(nums, 0, 0)

File: Leetcode/1863. Sum of All Subset XOR Totals.cpp
class Solution {
private:
    int helper(vector<int> &nums, int i, int N, int XOR) {
        return i == N ? XOR : helper(nums, i + 1, N, XOR) + helper(nums, i + 1, N, XOR ^ nums[i]);
    }
public:
    int subsetXORSum(vector<int>& nums) {
        // either pick it or leave it and add to sum
        return helper(nums, i, nums.size(), 0);
    }
};